components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -2414,5 +2421,81 @@ public static FederatedAssociationManager getFederatedAssociationManager() throw
         }
         return federatedAssociationManager;
     }
-}
 
+    /**
+     * Retrieves the unique user id of the given username. If the unique user id is not available, generate an id and
+     * update the userid claim in read/write userstores.
+     *
+     * @param tenantDomain    tenant domain of the user
+     * @param userStoreDomain userstore of the user
+     * @param username        username
+     * @return unique user id of the user
+     * @throws UserSessionException
+     */
+    public static String resolveUserIdFromUsername(String tenantDomain, String userStoreDomain, String username) throws
+            UserSessionException {
+
+        if (StringUtils.isEmpty(tenantDomain)) {
+            throw new UserSessionException("User tenant domain cannot be empty.");
+        }
+        try {
+            UserStoreManager userStoreManager = getUserStoreManager(tenantDomain, userStoreDomain);
+            try {
+                if (userStoreManager instanceof AbstractUserStoreManager) {
+                    String userId = ((AbstractUserStoreManager) userStoreManager).getUserIDFromUserName(username);
+                    if (StringUtils.isBlank(userId) && !userStoreManager.isReadOnly()) {
```

Comment:
Can you add a comment with the reason for this check?

Reply:
Added

Reply:
Shall we change the comment like below?
If the user id is not present in the userstore, we need to add it to the userstore. But if the userstore is read-only, we cannot add the id and will be throwing an error.